authenticationHelper.js

```diff
@@ -124,156 +85,106 @@ exports.handleOauthCallback = function(request, response) {
+      response.render('error', {
+        message: "Invalid request with state " + receivedState
+      });
     }
   });
 }
 
-function invalidateProcessedState(processedState, code, response) {
-  redisClient.srem(config.REDIS_VALID_AUTHS, processedState, function(err, stateDeleted) {
-    if (err) {
-      log.info("Something wasn't right with redis.");
-      console.error("Unable to check set membership for state " + processedState);
-      throw err;
-    }
-
-    if (stateDeleted) {
-      getNewAccessToken(code, response);
-    } else {
-      log.error('Unable to delete state ' + processedState + ' from redis');
-    }
-  })
-}
-
-function getNewAccessToken(code, response) {
-  request
-    .get("https://slack.com/api/oauth.access?" + "client_id=" + config.authClientId +
-      "&client_secret=" + config.authClientSecret + "&code=" + code)
-    .end(function(err, res) {
-      if (res.ok && res.body && (res.body.ok === true)) {
-        return saveAccessToken(res.body.access_token, response);
+function getNewAccessToken(code, expectedUserId, response) {
+  slackApi.oauth.access({
+      client_id: config.authClientId,
+      client_secret: config.authClientSecret,
+      code: code
+    })
+    .then(function(res) {
+      if (res.ok) {
+        return saveAccessToken(res.access_token, expectedUserId, response);
       } else {
         response.statusCode = 400;
-        //TODO: add handlebar template for this error
-        return response.end("Something went wrong with authentication.");
+        return response.render('error', {
+          message: "Something went wrong with authentication."
+        });
       }
+    })
+    .catch(function(err) {
+      throw err;
     });
 }
 
 //TODO: add a slack command to revoke user's auth token
 //      this will clear any queued actions
 //      and clear user's saved accessToken
-function saveAccessToken(accessToken, response) {
-  request
-    .get("https://slack.com/api/auth.test?" + "token=" + accessToken)
-    .end(function(err, res) {
+function saveAccessToken(accessToken, expectedUserId, response) {
+  slackApi.auth.test({
+      token: accessToken
+    })
+    .then(function(res) {
       if (res.ok) {
-        var userId = res.body.user_id;
-        redisClient.hset(userId, "accessToken", accessToken, function(err, res) {
-          if (err) {
-            log.info("Something wasn't right with redis.");
-            console.error("Unable to save accessToken for userId " + userId + " to redis");
-            throw err;
-          }
-          log.info("Updated user " + userId + "'s access token to " + accessToken);
-          displayAuthSuccessPage(accessToken, userId, response);
-        });
-      } else {
-        log.info("Could not query oauth.test for token: " + accessToken);
+        var userId = res.user_id;
+        var userName = res.user;
+        if (userId === expectedUserId) {
+          dao.saveAccessTokenForUser(accessToken, userId, function(res) {
```

Comment:
This would be a good place to promisify this function. Then you can simply chain the promises and do all of the error handling at the end.